dsl/src/main/scala/org/http4s/dsl/impl/Path.scala

```diff
@@ -166,12 +166,32 @@ object /: {
     }
 }
 
-protected class PathVar[A](cast: String => Try[A]) {
+/** Abstract extractor of a path variable:
+  * {{{
+  *   enum Color:
+  *     case Red, Green, Blue
+  *
+  *   val ColorPath = new PathVar(str => Try(Color.valueOf(str)))
+  *
+  *   Path("/Green") match {
+  *     case Root / ColorPath(color) => ...
+  * }}}
+  */
+class PathVar[A] private[impl] (cast: String => Option[A]) {
   def unapply(str: String): Option[A] =
-    if (!str.isEmpty)
-      cast(str).toOption
-    else
-      None
+    if (str.nonEmpty) cast(str)
+    else None
+}
+
+object PathVar {
+  def pure[A](cast: String => A): PathVar[A] =
+    new PathVar(str => Option(cast(str)))
```

Comment:
Any reason to expect a `null` value here? We can just use `Some` otherwise

Reply:
Have been burned by nulls before so I prefer Option to Some when receiving values outside my control.